x-pack/platform/plugins/shared/alerting/server/application/rule/methods/create/create_rule.test.ts

```diff
@@ -346,7 +346,9 @@ describe('create()', () => {
   });
 
   test('creates an rule', async () => {
-    const data = getMockData();
+    const data = getMockData({
```

Comment:
I had to add a dashboards property in the mock data to make my test work. Without it I was getting `dashboards: undefined` and I don't know where this undefined comes from. Could somebody from response ops team help here?


<img width="1031" alt="Screenshot 2025-03-19 at 09 29 56" src="https://github.com/user-attachments/assets/057d6541-09a6-46a5-8d21-abcd3475e4a7" />

Reply:
Ok I think I am going somewhere. It fails when it tries to validate the rule [here](https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/alerting/server/application/rule/methods/create/create_rule.ts#L268). I am gonna see how I can properly validate the data. I would appreciate your feedback and some best practices on how to add new keys in the rule Saved Object

Reply:
I have this PR as a guideline https://github.com/elastic/kibana/pull/190019, so maybe @JiaweiWu you can help me out here?

Reply:
Ι am still waiting for CI to report the status, but it looks like I managed to fix failing tests (locally) in this [commit](https://github.com/elastic/kibana/pull/209668/commits/c4c75d9af65b7ae9d585680007717dfe2456c83a)